libs/SFMLSystems/InputManagement.cpp

```diff
@@ -34,28 +34,37 @@ void InputManagement::run(World &world)
                 if (entityPtr->getComponent<KeyboardInputComponent>().keyboardMapActions.contains(event.key.code)
                     && entityPtr->contains<AllowMouseAndKeyboardComponent>())
                     entityPtr->getComponent<ActionQueueComponent>().actions.push(
-                        entityPtr->getComponent<KeyboardInputComponent>().keyboardMapActions[event.key.code].first);
+                        entityPtr->getComponent<KeyboardInputComponent>().keyboardMapActions[event.key.code]);
             };
             std::for_each(Inputs.begin(), Inputs.end(), keyPressed);
         }
+        if (event.type == sf::Event::KeyReleased) {
+            auto keyReleased = [event](std::shared_ptr<ecs::Entity> entityPtr) {
+                if (entityPtr->getComponent<KeyboardInputComponent>().keyboardMapActions.contains(event.key.code)
+                    && entityPtr->contains<AllowMouseAndKeyboardComponent>()) {
+                    entityPtr->getComponent<ActionQueueComponent>().actions.push(
+                        std::make_pair<ActionQueueComponent::inputAction_e, float>(ActionQueueComponent::inputAction_e(entityPtr->getComponent<KeyboardInputComponent>().keyboardMapActions[event.key.code].first), 0));
+                }
+            };
+            std::for_each(Inputs.begin(), Inputs.end(), keyReleased);
+        }
         if (event.type == sf::Event::MouseButtonPressed) {
             auto mouseButtonPressed = [event](std::shared_ptr<ecs::Entity> entityPtr) {
                 if (entityPtr->getComponent<MouseInputComponent>().MouseMapActions.contains(event.mouseButton.button)
-                    && entityPtr->contains<AllowMouseAndKeyboardComponent>())
-                    entityPtr->getComponent<ActionQueueComponent>().actions.push(
-                        entityPtr->getComponent<MouseInputComponent>().MouseMapActions[event.mouseButton.button].first);
+                    && entityPtr->contains<AllowMouseAndKeyboardComponent>()) {}
+                    // entityPtr->getComponent<ActionQueueComponent>().actions.push(
+                    //     entityPtr->getComponent<MouseInputComponent>().MouseMapActions[event.mouseButton.button]);
```

Comment:
Comment ?

Reply:
(I'm Thomas) I can remove them, but I will reimplement it in part 2 because it will be more efficient

Reply:
Np